File: javascript/rna-transcription/rna-transcription.js
/** INPUT: a DNA strand (string)
 *  OUTPUT: the input dna_strand converted to an RNA strand (string)
 *  ASSUMES: dna_strand is a valid DNA strand. */
export const toRna = (dna_strand) => {

  let rna_strand = '';
  for (const nucleotide of dna_strand) {
    rna_strand += nucleotide_translations[nucleotide];
  }

  return rna_strand;
};

const nucleotide_translations = { 'G': 'C'
                                , 'C': 'G'
                                , 'T': 'A'
                                , 'A': 'U' };

File: java/rna-transcription/src/main/java/RnaTranscription.java
class RnaTranscription {

    String transcribe(String dnaStrand) {
        char[] nucleotides = dnaStrand.toCharArray();
        int len = nucleotides.length;
        for (int i = 0; i < len; i++)
            nucleotides[i] = transcribe(nucleotides[i]);
        return new String(nucleotides);
    }

    char transcribe(char nucleotide) {
        switch (nucleotide) {
            case 'G':
                return 'C';
            case 'C':
                return 'G';
            case 'T':
                return 'A';
            case 'A':
                return 'U';
            default:
                throw new IllegalArgumentException("DNA strand contains invalid nucleotide.");
        }
    }
}